src/main/docs/guide/httpServer/byteBody.adoc

```diff
@@ -0,0 +1,134 @@
+Micronaut HTTP server version 4.5.0 introduced a new, more advanced API to access the bytes of an incoming request.
+When a api:io.micronaut.http.HttpRequest[] implements api:io.micronaut.http.ServerHttpRequest[], the new `byteBody()`
+method returns an api:io.micronaut.http.body.ByteBody[] with buffered, reactive and blocking APIs.
+
+When an HTTP request comes in, it starts out with an unparsed and unclaimed stream of bytes as its `byteBody()`. After
+all request filters have run, typically an argument binder matching the `@Body` parameter of the controller will
+"claim" the `byteBody()` and e.g. parse the JSON. Finally, the body is closed at the end of the request lifecycle,
+discarding any data if it has not been claimed by the argument binder.
+
+=== Primary operations
+
+`ByteBody` itself does not offer direct access to the data. To begin processing, there must be a _primary
+operation_ that converts the body into another form that can be used in the application programming model.
+
+A normal `ByteBody` has two groups of streaming primary operations. `toInputStream()` gives access to the body
+as a regular `InputStream`. The `toByteArrayPublisher()` and `toByteBufferPublisher()` methods return a reactive stream
+of byte arrays or `ByteBuffer`s.
+
+WARNING: `InputStream` is blocking API, and the netty event loop must never be blocked. If you wish to read from the
+body using an `InputStream`, take care to do so only on another thread, or to annotate your filter with
+`@ExecuteOn(TaskExecutors.BLOCKING)`.
+
+If you need full access to the body, the `buffer()` method returns an `ExecutionFlow` that completes with an
```

Comment:
is `ExecutionFlow` now public API? Currently it is annotated with `@Internal`? Is it ready to be public API?

Reply:
oh i forgot about this. @dstepanov 

I talked with Denis about this a few weeks ago and he wasn't against it iirc. But we may want to only make some of the methods public, the API is pretty big atm.

Another alternative is to have a second method that returns CompletionStage, and make the ExecutionFlow one Internal and default. That way we can still have the perf benefit of ExecutionFlow without exposing anything.

Reply:
ive moved to CompletableFuture